### PR TITLE
Update cdms table

### PR DESCRIPTION
**Pull request by keflavich**
The CDMS catalog directory table has been updated.  This PR updates our cached file & implements a tool for future updating & makes it possible to use the live, remote version.  The only reason this didn't exist previously was my own laziness; the formatting was weird b/c of html stuff mixed in with the table and I had just hand hacked it to fit.  Turns out it's pretty easy to read the table with `ascii.read`, just took some figuring.

### CODE REVIEW

**Comment by keflavich** [comment]
@bsipocz this is ready for review.  lmk if you think more tests are needed

**Comment by bsipocz** [comment]
The docs build issue is a genuine one.

File: astroquery/linelists/cdms/core.py [hunk 1]
```diff
@@ -277,7 +277,8 @@ def _parse_result(self, response, *, verbose=False):
 
         return result
 
-    def get_species_table(self, *, catfile='catdir.cat'):
+    def get_species_table(self, *, catfile='catdir.cat', use_cached=True,
+                          catfile_url='https://cdms.astro.uni-koeln.de/classic/entries/partition_function.html'):
```
Comment (bsipocz) [inline astroquery/linelists/cdms/core.py:281]: I wonder whether this should be in the config, or here?

File: astroquery/linelists/cdms/core.py [hunk 2]
```diff
@@ -299,8 +300,11 @@ def get_species_table(self, *, catfile='catdir.cat'):
 
         """
 
-        result = ascii.read(data_path('catdir.cat'), format='csv',
-                            delimiter='|')
+        if use_cached:
+            result = ascii.read(data_path(catfile), format='fixed_width',
+                                delimiter='|')
```
Comment (bsipocz) [inline astroquery/linelists/cdms/core.py:305]: we allow long lines, you can leave it in one

File: astroquery/linelists/cdms/core.py [hunk 3]
```diff
@@ -390,9 +394,28 @@ def find(self, st, flags):
...
     lookuptable = Lookuptable(dictionary)  # apply the class above
 
     return lookuptable
+
+
+def retrieve_catfile(url='https://cdms.astro.uni-koeln.de/classic/entries/partition_function.html'):
+    """
+    Simple retrieve index function
+    """
+    import requests
```
Comment (bsipocz) [inline astroquery/linelists/cdms/core.py:409]: do the import on the top, we only do this type when it's about an optional dependency (and in practice `requests` is very likely imported already, so there is no real overhead doing it at the top)

File: astroquery/linelists/cdms/core.py [hunk 4]
```diff
@@ -390,9 +394,28 @@ def find(self, st, flags):
...
+    resp = requests.get(url)
```
Comment (bsipocz) [inline astroquery/linelists/cdms/core.py:410]: nitpick, but use full words for variables when it makes sense

File: astroquery/linelists/cdms/data/catdir.cat [hunk 5]
```diff
@@ -1,1089 +1,1198 @@
-   TAG |  NAME                  |      NLINE   |  lg(Q(1000))  |  lg(Q(500))   |  lg(Q(300))  |   lg(Q(225))  |   lg(Q(150))   |   lg(Q(75))  |  lg(Q(37.5))   |  lg(Q(18.75))  |  lg(Q(9.375)) |   lg(Q(5.000))  |  lg(Q(2.725))
```
Comment (bsipocz) [inline astroquery/linelists/cdms/data/catdir.cat:1]: There has been a lot of back and forth with this file, please squash those changes out into one commit touching the datafile.

File: docs/linelists/cdms/cdms.rst [hunk 6]
```diff
@@ -128,22 +128,22 @@ Reference/API section.
 Looking Up More Information from the catdir.cat file
 ------------------------------------------------------
 
-If you have found a molecule you are interested in, the TAG field in the
+If you have found a molecule you are interested in, the ``tag`` field in the
```
Comment (bsipocz) [inline docs/linelists/cdms/cdms.rst:131]: is it a field or rather a column?

**Review by bsipocz — CHANGES_REQUESTED** [review]
This needs a changelog, and the docs build and tests fixed. 

Maybe some test coverage would be nice, too as the new functionality is not tested anywhere. If you think its unnecessarily big download, then we can package up a much smaller test file in data and use its github url.

File: astroquery/linelists/cdms/core.py (hunk 1 above)
Comment (keflavich) [reply astroquery/linelists/cdms/core.py:281]: probably config

File: docs/linelists/cdms/cdms.rst (hunk 6 above)
Comment (keflavich) [reply docs/linelists/cdms/cdms.rst:131]: column

File: astroquery/linelists/cdms/data/catdir.cat (hunk 5 above)
Comment (keflavich) [reply astroquery/linelists/cdms/data/catdir.cat:1]: can we just squash at the end?

**Comment by keflavich** [comment]
Only remaining red is code coverage.

**Comment by bsipocz** [comment]
> Only remaining red is code coverage.

Ok, could you add a changelog, and squash the datafile modifications?

**Review by bsipocz — APPROVED** [review]

**Comment by bsipocz** [comment]
Thanks @keflavich!

**Pull request merged**